src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,289 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorscaleAttrs = require('../../components/colorscale/attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var surfaceAtts = require('../surface/attributes');
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+function makeSliceAttr(axLetter) {
+    return {
+        editType: 'calc',
+        show: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: false,
+            editType: 'calc',
+            description: [
+                'Determines whether or not slice planes about the', axLetter,
+                'dimension are drawn.'
+            ].join(' ')
+        },
+        locations: {
+            valType: 'data_array',
+            dflt: [],
+            role: 'info',
+            editType: 'calc',
+            description: [
+                'Specifies the location(s) of slices on the axis [0, n].',
+                'When not locations specified slices would be created for',
+                'all (0, n) i.e. except start and end caps.'
+            ].join(' ')
+        },
+        fill: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            max: 1,
+            dflt: 1,
+            editType: 'calc',
+            description: 'Sets the fill ratio of the interior slices.'
+        }
+    };
+}
+
+function makeCapAttr(axLetter) {
+    return {
+        editType: 'calc',
+        show: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: true,
+            editType: 'calc',
+            description: [
+                'Determines whether or not cap planes on both sides of the', axLetter,
+                'dimension are drawn. Note: it is possible to use `slices` in order to',
+                'display a cap only on one side.'
+            ].join(' ')
+        },
+        fill: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            max: 1,
+            dflt: 1,
+            editType: 'calc',
+            description: 'Sets the fill ratio of the caps.'
+        }
+    };
+}
+
+
+module.exports = extendFlat({
```

Comment:
Would be nice to use `overrideAll(/**/, 'calc', 'nested')` like surface does:

https://github.com/plotly/plotly.js/blob/cc4597257b411688b2bbc6dbe90f8a6e78cc96c8/src/traces/surface/attributes.js#L17